analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -43,9 +44,77 @@ protected override ProgramState[] Process(SymbolicContext context, IInvocationOp
         {
             state = state.ResetFieldConstraints();
         }
-        return invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace))
-            ? ProcessStringIsNullOrEmpty(context, invocation)
-            : new[] { state };
+        if (invocation.TargetMethod.ContainingType.Is(KnownType.System_Linq_Enumerable))
```

Comment:
What about [System.Linq.Queryable](https://learn.microsoft.com/dotnet/api/system.linq.queryable). It has the same methods and the IQueryProvider implementations are expected to behave the same way Enumerable does.